### PR TITLE
specify `appProtocol` for services that use https

### PR DESCRIPTION
**Pull request by a7i**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

For clusters that use istio, you must specify port name following format $name-$protocol or use appProtocol.

Istio Docs: https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/#explicit-protocol-selection

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
add `appProtocol` to manifests for kind service
```

### CODE REVIEW

File: artifacts/deploy/karmada-search.yaml [hunk 1]
```diff
@@ -77,7 +77,8 @@ metadata:
     apiserver: "true"
 spec:
   ports:
-    - port: 443
+    - appProtocol: https
+      port: 443
```
Comment (zhzhuang-zju) [inline artifacts/deploy/karmada-search.yaml:81]: Karmada has different installation methods, and the directories `artifact/deploy` and `charts/karmada/templates` correspond to two different installation methods. Should the configurations of different installation methods be the same? If so, you can refer to https://github.com/karmada-io/karmada/issues/4191 to find the configuration files for different installation methods.

**Review by zhzhuang-zju — COMMENTED** [review]
Hi~ @a7i  thanks for your work! Review is ongoing, and my viewpoint just for reference only.

**Comment by zhzhuang-zju** [comment]
Hi~ Ask out of curiosity, have you combined Karmada with Istio? If so, could you share your use cases and experiences? Thank you.

File: artifacts/deploy/karmada-search.yaml [hunk 2]
```diff
@@ -77,7 +77,8 @@ metadata:
     apiserver: "true"
 spec:
   ports:
-    - port: 443
+    - appProtocol: https
```
Comment (RainbowMango) [inline artifacts/deploy/karmada-search.yaml:80]: Is this also useful when working with other system other than Istio? In other words, if another system requires another appProtocol?
(I'm not an expert in this area, maybe stupid question)

File: artifacts/deploy/karmada-search.yaml (hunk 1 above)
Comment (a7i) [reply artifacts/deploy/karmada-search.yaml:81]: `appProtocol` is well supported field regardless of the install method. shouldn't these two be in sync as much as possible?

**Comment by a7i** [comment]
> Hi~ Ask out of curiosity, have you combined Karmada with Istio? If so, could you share your use cases and experiences? Thank you.

@zhzhuang-zju we do not have karmada in the mesh to prevent dependencies on cluster critical resources. With that being said, istio requires `appProtocol` or port `name` to follow this [Protocol Selection](https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/). We have opa-gatekeeper policies in place to ensure that if a Service exposes port 80/443 that it specifies `appProtocol` or follow correct naming convention. Otherwise, this breaks routing for ALL services listening on 80/443, causing a cluster-level outage. See related issue: https://github.com/istio/istio/issues/47088

File: artifacts/deploy/karmada-search.yaml (hunk 2 above)
Comment (a7i) [reply artifacts/deploy/karmada-search.yaml:80]: Also not an expert as we only use Istio :)

File: artifacts/deploy/karmada-search.yaml (hunk 1 above)
Comment (zhzhuang-zju) [reply artifacts/deploy/karmada-search.yaml:81]: Yes, they should be in sync as much as possible. So you may need to make the same changes in different installation methods.

File: artifacts/deploy/karmada-search.yaml (hunk 2 above)
Comment (zhzhuang-zju) [reply artifacts/deploy/karmada-search.yaml:80]: > In other words, if another system requires another appProtocol? (I'm not an expert in this area, maybe stupid question)

I think the configuration files we provide here should be universal and independent of specific usage. Can we use them as a practice to adapt Karmada to Istio and showcase it in tutorials or other documents?

**Comment by a7i** [comment]
We ended up adding a kustomize patch to do this. I think it's a good change to add upstream since `appProtocol` is HTTPS

**Pull request closed without merging**